docs/core/extensions/sslstream-best-practices.md

```diff
@@ -0,0 +1,107 @@
+---
+title: TLS/SSL best practices
+description: Learn best practices to using SslStream in .NET
+author: rzikm
+ms.author: rzikm
+ms.date: 10/25/2022
+---
+
+# TLS/SSL best practices
+
+TLS (Transport Layer Security) is a cryptographic protocol designed to secure communication between two computers over the internet. The TLS protocol is exposed in .NET via the <xref:System.Net.Security.SslStream> class.
+
+This article presents best practices for setting up secure communication between client and server and assumes .NET Core version 3.1 or later. For best practices for .NET Framework, see [Transport Layer Security (TLS) best practices with the .NET Framework](/dotnet/framework/network-programming/tls).
+
+## Selecting TLS version
+
+While it is possible to specify the version of the TLS protocol to be used via the SslProtocols property, it is recommended to defer to the operating system settings by using <xref:System.Net.Security.SslProtocols.None> value (this is the default).
+
+Deferring the decision to the OS automatically uses the most recent version of TLS available and lets the application pick up changes after OS upgrades. The operating system may also prevent use of TLS versions which are no longer considered secure.
+
+## Selecting cipher suites
+
+SslStream allows users to specify which cipher suites can be negotiated by the TLS handshake via the <xref:System.Net.Security.CipherSuitesPolicy> class. As with TLS versions, we recommend letting the operating system decide which are the best cipher suites to negotiate with and, therefore, we recommend avoiding the use of <xref:System.Net.Security.CipherSuitesPolicy>.
+
+Note that <xref:System.Net.Security.CipherSuitesPolicy> is not available on all platforms supported by .NET.
+
+## Specifying a Server Certificate
+
+When authenticating as a server, <xref:System.Net.Security.SslStream> requires an <xref:System.Security.Cryptography.X509Certificates.X509Certificate> instance to use. There are multiple ways how the server certificate can be passed to <xref:System.Net.Security.SslStream>:
```

Comment:
I would also mention that the certificate should have private key. We will search through stores for it if it does not bat that is problematic IMHO. I think @bartonjs would also prefer to use `X509Certificate2` everywhere we can. 
On Windows the certificate/key must be in store or persistent storage so `schanel` can access it. While this is implementation details, I would probably mention it. 
Perhaps @bartonjs can recommend some import/create fragment for anybody loading it from PEM of pfx.

Reply:
> On Windows the certificate/key must be in store or persistent storage so schanel can access it.

Does it really? I thought this holds only for the intermediate certificates and the leaf certificate does not have to be in the store.

Reply:
I was primarily thinking about https://github.com/dotnet/runtime/issues/23749
Getting them to store is one way how to fix it. When loaded from file, there is some trickery about the flags. 
There is bunch of linked issues as people seems to subtle on this regularly. We can also move it down to the debug section. I'm not sure if conceptual doc is good dumping ground for platform caveats.